contrib/experimental/great_expectations_experimental/expectations/expect_queried_column_pair_values_to_be_both_filled_or_null.py

```diff
@@ -0,0 +1,149 @@
+
+     It checks if 2 columns are aligned - the values of each row need to either be both empty or filled.
+     The expectation will fail if there's at least one row where one column is filled and the other isn't.
+
+    Args:
+    template_dict: dict with the following keys: \
+        column_a (str): first column name, to compare values against column_b
+        column_b (str): second column name, to compare values against column_a
+
+    Returns:
+            None
+    """
+
+    metric_dependencies = ("query.template_values",)
+
+    query = """
+        SELECT
+            COUNT(1)
+        FROM
+            {active_batch}
+        WHERE
+            ({column_a} is not null and {column_b} is null)
+            OR
+            ({column_a} is null and {column_b} is not null)
+            """
+
+    success_keys = (
+        "template_dict",
+        "query",
+    )
+
+    domain_keys = ("batch_id", "row_condition", "condition_parser")
+
+    default_kwarg_values = {
+        "result_format": "BASIC",
+        "include_config": True,
+        "catch_exceptions": False,
+        "meta": None,
+        "query": query,
+    }
+
+    def validate_configuration(
+        self, configuration: Optional[ExpectationConfiguration]
+    ) -> None:
+        """
+        Validates that a configuration has been set, and sets a configuration if it has yet to be set. Ensures that
+        necessary configuration arguments have been provided for the validation of the expectation.
+
+        Args:
+            configuration (OPTIONAL[ExpectationConfiguration]): \
+                An optional Expectation Configuration entry that will be used to configure the expectation
+        Returns:
+            None. Raises InvalidExpectationConfigurationError if the config is not validated successfully
+        """
+        super().validate_configuration(configuration)
+        configuration = configuration or self.configuration
+
+        template_dict = configuration.kwargs.get("template_dict")
+        try:
+            assert isinstance(
+                template_dict, dict
+            ), "'template_dict' must be supplied as a dict"
+            assert all(
+                [
+                    "column_a" in template_dict,
+                    "column_b" in template_dict,
+                ]
+            ), "The following keys have to be in the template dict: column_a, column_b "
+        except AssertionError as e:
+            raise InvalidExpectationConfigurationError(str(e))
+
+    def _validate(
+        self,
+        configuration: ExpectationConfiguration,
+        metrics: dict,
+        runtime_configuration: dict = None,
+        execution_engine: ExecutionEngine = None,
+    ) -> Union[ExpectationValidationResult, dict]:
+        metrics = convert_to_json_serializable(data=metrics)
+        num_of_inconsistent_rows = list(
+            metrics.get("query.template_values")[0].values()
+        )[0]
```

Comment:
It would be good to check that the 0th index exists and output a more specific error here if this is not already caught and handled within a try/catch block elsewhere.